### PR TITLE
Validate AnimatePresence keys - warn if they are not unique

### LINKED ISSUE #166: Add error warning for multiple children of `AnimatePresence` with the same key.
Currently `AnimatePresence` doesn't warn if a component doesn't have a valid key. This is to allow toggling the visibility of single components without needing one.

We can either:
1) Require even for single-child examples and throw
2) Detect multiple of the same key if we have more than one child and throw

2 would miss the `Slideshow`- esque use-case where we have a single child with a changing key, but would enable the toggling of visibility of a single child without a key. I am leaning towards 1.

### PR DESCRIPTION
**Pull request by Andarist**
fixes #166

### CODE REVIEW

File: src/components/AnimatePresence.tsx [hunk 1]
```diff
@@ -79,8 +79,20 @@ function updateChildLookup(
     children: ReactElement<any>[],
     allChildren: Map<ComponentKey, ReactElement<any>>
 ) {
+    const seenChildren =
+        process.env.NODE_ENV !== "production" ? new Set<ComponentKey>() : null
```
Comment (Andarist) [inline src/components/AnimatePresence.tsx]: those env checks look slightly weird but they provide better DCE in production builds

File: src/components/AnimatePresence.tsx [hunk 2]
```diff
@@ -79,8 +79,20 @@ function updateChildLookup(
...
+
     children.forEach(child => {
         const key = getChildKey(child)
+
+        if (process.env.NODE_ENV !== "production" && seenChildren) {
+            if (seenChildren.has(key)) {
+                console.warn("AnimatePresence's children need unique keys.")
```
Comment (InventingWithMonster) [inline src/components/AnimatePresence.tsx]:
```
console.warn(`Children of AnimatePresence require unique keys. ${key} is a duplicate.`)
```

Comment (Andarist) [reply src/components/AnimatePresence.tsx]: adjusted

**Comment by InventingWithMonster** [comment]
Nice one thanks!

**Pull request merged**